xbmc/settings/DisplaySettings.cpp

```diff
@@ -548,23 +548,23 @@ void CDisplaySettings::ApplyCalibrations()
 void CDisplaySettings::UpdateCalibrations()
 {
   CSingleLock lock(m_critical);
-  for (size_t res = RES_DESKTOP; res < m_resolutions.size(); ++res)
+
+  // Add new (unique) resolutions
+  for (ResolutionInfos::const_iterator res(m_resolutions.cbegin() + RES_DESKTOP + 1); res != m_resolutions.cend(); ++res)
+    if (std::find_if(m_calibrations.cbegin(), m_calibrations.cend(),
+      [&](const RESOLUTION_INFO& info) { return StringUtils::EqualsNoCase(res->strMode, info.strMode); }) == m_resolutions.cend())
+        m_calibrations.push_back(*res);
```

Comment:
My suggestion for newlines:
```c++
  for (ResolutionInfos::const_iterator res(m_resolutions.cbegin() + RES_DESKTOP + 1); res != m_resolutions.cend(); ++res)
  {
    if (std::find_if(m_calibrations.cbegin(), m_calibrations.cend(),
      [&](const RESOLUTION_INFO& info)
      {
        return StringUtils::EqualsNoCase(res->strMode, info.strMode);
      }) == m_resolutions.cend())
    {
      m_calibrations.push_back(*res);
    }
  }
```

Reply:
Looks ugly still but is more readable on small displays

Reply:
@rechi what's your opinion? I don't really care, even I'm a friend of not too many lines....

Reply:
Current and suggested variant are okay for me. I don't care which one you use.